### PR TITLE
FE-2269 - react-transition-group update

### PR DESCRIPTION
**Pull request by dawidzarzycki**
# Description
    Update React-Transition-Group to the latest version
    
    Updated components and files:

    - Carousel
    - Show Edit Pod
    - Animation Menu Button
    - Dialog
    - Dialog Full Screen
    - Modal
    - Pages
    - Sidebar
    - other components based on Dialog and Modal
    - Table
    - Toast
    - Flash
    - Flash Legacy
    - document.js

    What has been changed:
    - replaced `CSSTransitionGroup` with `CSSTransition` and `TransitionGroup`
    - renamed css classes
    - updated tests and snapshots
    - removed unnecessary comments
    - added missing `propTypes` description
    - moved `propTypes` and `defaultProps` to the bottom of the components

### CODE REVIEW

**Comment by jamime** [comment]
fixes #2397

File: demo/views/pages/document/document.js [hunk 1]
```diff
@@ -44,32 +44,39 @@ class Document extends React.Component {
...
+        <div className='demo-document__content'>
+          <Highlight innerHTML>
             { marked(content) }
           </Highlight>
         </div>
       );
     } else {
-      spinner = <Row gutter="none"><Spinner columnAlign="center" size="small" className="demo-document__spinner"/></Row>;
+      spinner = (
+        <Row gutter='none'><Spinner
```
Comment (mkrds) [inline demo/views/pages/document/document.js]:
```suggestion
  <Row gutter='none'>
    <Spinner
      columnAlign='center'
      size='small'
      className='demo-document__spinner'
    />
  </Row>
```

File: src/components/carousel/carousel.component.js [hunk 2]
```diff
@@ -149,10 +148,17 @@ class BaseCarousel extends React.Component {
...
+  visibleSlide = () => {
     let index = this.state.selectedSlideIndex;
 
-    const visibleSlide = compact(React.Children.toArray(this.props.children))[index];
+    const visibleSlide = (
+      <CSSTransition
+        className='carbon-carousel__transition'
+        classNames={ this.transitionName() }
+        timeout={ { enter: TRANSITION_TIME, exit: TRANSITION_TIME } }
+      >{compact(React.Children.toArray(this.props.children))[index]}
```
Comment (mkrds) [inline src/components/carousel/carousel.component.js]: Formatting - put `compact...` in new row

File: demo/views/pages/document/document.js (hunk 1 above)
Comment (dawidzarzycki) [reply demo/views/pages/document/document.js]: Updated in https://github.com/Sage/carbon/pull/2496/commits/ac367dc8ea8d2c0b8524b08a78f38b0d13800f4e

File: src/components/carousel/carousel.component.js (hunk 2 above)
Comment (dawidzarzycki) [reply src/components/carousel/carousel.component.js]: Updated in https://github.com/Sage/carbon/pull/2496/commits/ac367dc8ea8d2c0b8524b08a78f38b0d13800f4e

**Pull request merged**